core/store/orm/utils.go

```diff
@@ -0,0 +1,29 @@
+package orm
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+)
+
+func GormTransaction(db *gorm.DB, fc func(tx *gorm.DB) error) (err error) {
```

Comment:
Is it duplicated from the one in `orm`? Can it be reconciled if so?

Reply:
It's identical, I just needed a version that wasn't a method on the `orm.ORM` struct.  We can call this from inside of that method to deduplicate.